powa--4.0.1--4.1.0.sql

```diff
@@ -0,0 +1,613 @@
+ALTER TYPE powa_statements_history_rate ADD ATTRIBUTE wal_fpi_per_sec double precision;
+ALTER TYPE powa_statements_history_rate ADD ATTRIBUTE wal_bytes_per_sec numeric;
+
+CREATE OR REPLACE FUNCTION powa_statements_history_div(
+    a powa_statements_history_record,
+    b powa_statements_history_record)
+RETURNS powa_statements_history_rate AS
+$_$
+DECLARE
+    res powa_statements_history_rate;
+    sec integer;
+BEGIN
+    res.sec = extract(EPOCH FROM (a.ts - b.ts));
+    IF res.sec = 0 THEN
+        sec = 1;
+    ELSE
+        sec = res.sec;
+    END IF;
+    res.calls_per_sec = (a.calls - b.calls)::double precision / sec;
+    res.runtime_per_sec = (a.total_exec_time - b.total_exec_time)::double precision / sec;
+    res.rows_per_sec = (a.rows - b.rows)::double precision / sec;
+    res.shared_blks_hit_per_sec = (a.shared_blks_hit - b.shared_blks_hit)::double precision / sec;
+    res.shared_blks_read_per_sec = (a.shared_blks_read - b.shared_blks_read)::double precision / sec;
+    res.shared_blks_dirtied_per_sec = (a.shared_blks_dirtied - b.shared_blks_dirtied)::double precision / sec;
+    res.shared_blks_written_per_sec = (a.shared_blks_written - b.shared_blks_written)::double precision / sec;
+    res.local_blks_hit_per_sec = (a.local_blks_hit - b.local_blks_hit)::double precision / sec;
+    res.local_blks_read_per_sec = (a.local_blks_read - b.local_blks_read)::double precision / sec;
+    res.local_blks_dirtied_per_sec = (a.local_blks_dirtied - b.local_blks_dirtied)::double precision / sec;
+    res.local_blks_written_per_sec = (a.local_blks_written - b.local_blks_written)::double precision / sec;
+    res.temp_blks_read_per_sec = (a.temp_blks_read - b.temp_blks_read)::double precision / sec;
+    res.temp_blks_written_per_sec = (a.temp_blks_written - b.temp_blks_written)::double precision / sec;
+    res.blk_read_time_per_sec = (a.blk_read_time - b.blk_read_time)::double precision / sec;
+    res.blk_write_time_per_sec = (a.blk_write_time - b.blk_write_time)::double precision / sec;
+    res.plans_per_sec = (a.plans - b.plans)::double precision / sec;
+    res.plantime_per_sec = (a.total_plan_time - b.total_plan_time)::double precision / sec;
+    res.wal_records_per_sec = (a.wal_records - b.wal_records)::double precision / sec;
+    res.wal_fpi_per_sec = (a.wal_fpi - b.wal_fpi)::double precision / sec;
+    res.wal_bytes_per_sec = (a.wal_bytes - b.wal_bytes)::double precision / sec;
+
+    return res;
+END;
+$_$
+LANGUAGE plpgsql IMMUTABLE STRICT;
+
+ALTER TABLE public.powa_statements_src_tmp RENAME total_time TO total_exec_time;
+ALTER TABLE public.powa_statements_src_tmp ADD plans bigint NOT NULL;
+ALTER TABLE public.powa_statements_src_tmp ADD total_plan_time double precision NOT NULL;
+ALTER TABLE public.powa_statements_src_tmp ADD wal_records bigint NOT NULL;
+ALTER TABLE public.powa_statements_src_tmp ADD wal_fpi bigint NOT NULL;
+ALTER TABLE public.powa_statements_src_tmp ADD wal_bytes numeric NOT NULL;
```

Comment:
A truncate just before would be safer, you may fail if the table isn't empty (same for 4.0.2 -> 4.1.0)

Reply:
Good catch!

Reply:
And fixed.